platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/SolrCloudClientFactory.java

```diff
@@ -0,0 +1,209 @@
+        return Failsafe.with(retryPolicy)
+                .with(EXECUTOR_SERVICE)
+                .onRetry((c, failure, ctx) -> LOGGER.debug(
+                        "Attempt {} failed to create Solr Cloud client ({}). Retrying again.",
+                        ctx.getExecutions(),
+                        collection))
+                .onFailedAttempt(failure -> LOGGER.debug(
+                        "Attempt failed to create Solr Cloud client (" + collection + ")",
+                        failure))
+                .onSuccess(client -> LOGGER.debug("Successfully created Solr Cloud client ({})",
+                        collection))
+                .onFailure(failure -> LOGGER.warn(
+                        "All attempts failed to create Solr Cloud client (" + collection + ")",
+                        failure))
+                .get(() -> createSolrCloudClient(zookeeperHosts, collection));
+    }
+
+    private static SolrClient createSolrCloudClient(String zookeeperHosts, String collection) {
+
+        CloudSolrClient client = new CloudSolrClient(zookeeperHosts);
+        client.connect();
+
+        try {
+            uploadCoreConfiguration(collection, client);
+        } catch (SolrFactoryException e) {
+            LOGGER.debug("Unable to upload configuration to Solr Cloud", e);
+            return null;
+        }
+
+        try {
+            createCollection(collection, client);
+        } catch (SolrFactoryException e) {
+            LOGGER.debug("Unable to create collection on Solr Cloud", e);
+            return null;
+        }
+
+        client.setDefaultCollection(collection);
+        return client;
+    }
+
+    private static void createCollection(String collection, CloudSolrClient client)
+            throws SolrFactoryException {
+        try {
+            CollectionAdminResponse response = new CollectionAdminRequest.List().process(client);
+
+            if (response == null || response.getResponse() == null || response.getResponse()
+                    .get("collections") == null) {
+                throw new SolrFactoryException("Failed to get a list of existing collections");
+            }
+
+            List<String> collections = (List<String>) response.getResponse()
+                    .get("collections");
+            if (!collections.contains(collection)) {
+                response = new CollectionAdminRequest.Create().setNumShards(SHARD_COUNT)
+                        .setMaxShardsPerNode(MAXIMUM_SHARDS_PER_NODE)
+                        .setReplicationFactor(REPLICATION_FACTOR)
+                        .setCollectionName(collection)
+                        .process(client);
+                if (!response.isSuccess()) {
+                    throw new SolrFactoryException(
+                            "Failed to create collection [" + collection + "]: "
+                                    + response.getErrorMessages());
+                }
+                isCollectionReady(client, collection);
```

Comment:
Do we need to do something with the boolean that's returned? What should we do if we can't verify that the collections are ready? Fail?